### ISSUE TITLE
Discussin: Might be able to re-use more built-in logic now

### ISSUE THREAD

**Issue by tjdevries** [haorenW1025/diagnostic-nvim#18]
Hey, just drive-by issue of saying you might be able to re-use more of the built-in stuff for diagnostics now as we have exposed more stuff in master.

Also wondering if there was more stuff we could expose that would be helpful to you in this plugin.

Thanks.

**Comment by haorenW1025** [comment]
Yeah actually I've been planing to of move some stuff out, but preserving backward compatibility while doing that is kind of hard...I'll start by moving some redundant stuff out and slowly integrate to the built-in stuff. Thanks for the reminder still:)

**Comment by clason** [comment]
I don't think you need to worry about backwards compatibility -- both neovim LSP and this plugin are in early access, so "here there be dragons" and all that... :)

(Maybe a fat disclaimer on top of the README to that effect might be helpful to manage expectations, though -- people tend to get excited when they see shiny new things!)

**Comment by haorenW1025** [comment]
@clason Make sense. Maybe I'll open a new branch and keep up with the latest built-in functions. Eventually it'll be merged into the master branch.

**Comment by aktau** [comment]
Which functions are we talking about here?
